woocommerce/woocommerce-android
# REST API: Jetpack setup event tracking

Fixes #8702.

This PR adds event tracking for a Jetpack setup flow, including the login, plugin installation and activation events. 

It'd be very time-consuming to test every possible branch events, so I'd suggest reviewing the code and checking the event placement (especially failures) and then testing the nominal flow with email/password.

**To test:** 
1. Start with a self-hosted site without Jetpack
2. Log in, then tap on the Jetpack banner
3. Go through the setup flow and verify that the Jetpack setup tracking events go something like this:

```
🔵 Tracked: feature_jetpack_benefits_banner, Properties: {"action":"tapped","blog_id":0,"is_wpcom_store":false,"is_debug":true}
🔵 Tracked: jetpack_benefits_login_button_tapped, Properties: {"blog_id":0,"is_wpcom_store":false,"is_debug":true}
🔵 Tracked: jetpack_setup_connection_check_completed, Properties: {"is_already_connected":false,"requires_connection_only":false,"blog_id":0,"is_wpcom_store":false,"is_debug":true}
🔵 Tracked: jetpack_setup_login_flow, Properties: {"step":"email_address","tap":"submit","blog_id":0,"is_wpcom_store":false,"is_debug":true}
🔵 Tracked: jetpack_setup_login_flow, Properties: {"step":"password","tap":"submit","blog_id":0,"is_wpcom_store":false,"is_debug":true}
🔵 Tracked: jetpack_setup_login_completed, Properties: {"blog_id":0,"is_wpcom_store":false,"is_debug":true}
🔵 Tracked: jetpack_setup_completed, Properties: {"blog_id":0,"is_wpcom_store":false,"is_debug":true}
🔵 Tracked: jetpack_setup_flow, Properties: {"step":"all_done","tap":"go_to_store","blog_id":0,"is_wpcom_store":false,"is_debug":true}
🔵 Tracked: jetpack_setup_synchronization_completed, Properties: {"blog_id":0,"is_wpcom_store":false,"is_debug":true}
```

**For reference:**
<img width="1091" alt="image" src="https://user-images.githubusercontent.com/1522856/230508988-94a6bb4e-98eb-4441-a67e-bd4e5c2ddb60.png">
